src/client/Client.cc

```diff
@@ -11640,8 +11644,7 @@ int Client::_setxattr(Inode *in, const char *name, const void *value,
 	if (ret < 0)
 	  return ret;
 	if (ret == 0) {
-	  value = NULL;
-	  size = 0;
+	  return _removexattr(in, name, perms);
```

Comment:
http://qa-proxy.ceph.com/teuthology/sidharthanup-2020-06-30_09:25:08-fs-wip-fix-setxattr-client5-distro-basic-smithi/5191589/teuthology.log is the log of the run where it fails (with the current changes).
So it seems that even with the latest change it's still breaking the pjd suite acl tests. From looking at the logs it seems that the MDS is returning ENODATA in response to the _removexattr call here . The mds code that does this is here: https://github.com/ceph/ceph/blob/master/src/mds/Server.cc#L5947 . Now before this PR, the xattr removal (that's supposed to happen here due to the size and value being set to 0 and NULL respectively) was done implicitly as part of the setxattr call and on the mds side it is handled here: https://github.com/ceph/ceph/blob/master/src/mds/Server.cc#L5893-L5895 . There is no check and hence no ENODATA response. Hence the tests did pass in that case. 

@batrick @jtlayton @zheng How do you propose we change the code here?

Reply:
You could just ignore `ENODATA` in this code path.

Reply:
Yeah, agreed. I don't see that as a reason to resurrect CEPH_XATTR_REMOVE; just fix up the error handling.

Reply:
Perfect. Thanks @batrick @jtlayton. Let me rerun the tests with the change.